### PR TITLE
feat: add checkmarks for satisfied password requirements

### PR DESCRIPTION
**Pull request by ni-jessica**

### CODE REVIEW

**Comment by ni-jessica** [comment]
https://github.com/csirianni/private-data-lookup/assets/98488267/beeb5e5a-2eef-414a-a6ec-37109d1bbd07

Ignore the error I wasn't running backend lol

File: frontend/app/page.tsx [hunk 1]
```diff
@@ -254,16 +267,35 @@ export default function SignUp() {
...
+                                    rules={[
+                                        "minLength",
+                                        "lowercase",
+                                        "capital",
+                                        "number",
+                                        "specialChar",
+                                        "match",
+                                    ]}
+                                    messages={{
+                                        specialChar: "Password has a special character.",
```
Comment (stellaljung) [inline frontend/app/page.tsx:282]: **question**: Do we want to list the special characters that can be used here that match our password regex? So "!#$%&()*+,-./:;<=>?@[]^_`{|}~"

File: frontend/app/page.tsx [hunk 2]
```diff
@@ -254,16 +267,35 @@ export default function SignUp() {
...
                             >
-                                <p className="-mb-2">Password must contain:</p>
-                                <ul>
-                                    <li>At least 10 characters</li>
-                                    <li>At least 1 lowercase character</li>
-                                    <li>At least 1 uppercase character</li>
-                                    <li>At least 1 number</li>
-                                    <li>At least 1 special character</li>
-                                </ul>
+                                <PasswordChecklist
```
Comment (stellaljung) [inline frontend/app/page.tsx:272]: **praise**: Good use of react components! It makes the code so clean and easy to read.

**Review by stellaljung — CHANGES_REQUESTED** [review]

**Review by stellaljung — APPROVED** [review]

**Comment by csirianni** [comment]
Screenshot 2023-12-24 at 5 16 09 PM

**Pull request merged**